src/client/components/chart-line/chart-line.tsx

```diff
@@ -32,109 +32,107 @@ export interface ChartLineProps {
   scaleX: (v: any) => number;
   scaleY: (v: any) => number;
   color: string;
+  dashed?: boolean;
   showArea?: boolean;
   hoverRange?: PlywoodRange;
 }
 
-export interface ChartLineState {
-}
-
-export class ChartLine extends React.Component<ChartLineProps, ChartLineState> {
-
-  render() {
-    var { stage, dataset, getY, getX, scaleX, scaleY, color, showArea, hoverRange } = this.props;
-    if (!dataset || !color) return null;
-
-    var dataPoints: Array<[number, number]> = [];
-    var hoverDataPoint: [number, number] = null;
+export const ChartLine: React.SFC<ChartLineProps> = ({ stage, dataset, getY, getX, scaleX, scaleY, color, showArea, hoverRange, dashed }) => {
+  if (!dataset || !color) return null;
 
-    var ds = dataset.data;
-    for (var i = 0; i < ds.length; i++) {
-      var datum = ds[i];
-      var range = getX(datum) as PlywoodRange;
+  let dataPoints: Array<[number, number]> = [];
+  let hoverDataPoint: [number, number] = null;
 
-      if (!range || !Range.isRange(range)) return null; // !range => Incorrect data loaded, !Range.isRange => temp solution for non-bucketed reaching here
+  const ds = dataset.data;
+  for (let i = 0; i < ds.length; i++) {
+    const datum = ds[i];
+    const range = getX(datum) as PlywoodRange;
 
-      var rangeMidpoint = (range as NumberRange | TimeRange).midpoint();
-      var measureValue = getY(datum);
+    const incorrectData = !range; // !range => Incorrect data loaded, !Range.isRange => temp solution for non-bucketed reaching here
```

Comment:
please add // TODO tag, ideally with linked issue

Reply:
Ugh, honestly I'm just moving this line around. Maybe it's time to dive in and try to figure out what's happening here.

Reply:
Oh, I see - please please ignore my comment